src/controllers/dynakube/dynakube_controller.go

```diff
@@ -217,23 +221,26 @@ func (controller *DynakubeController) reconcileDynaKube(ctx context.Context, dkS
 		upd, err = oneagent.NewOneAgentReconciler(
 			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.HostMonitoringFeature,
 		).Reconcile(ctx, dkState)
-		if dkState.Error(err) || dkState.Update(upd, defaultUpdateInterval, "infra monitoring reconciled") {
+		if dkState.Error(err) {
 			return
 		}
+		dkState.Update(upd, defaultUpdateInterval, "infra monitoring reconciled")
```

Comment:
`dkState.Update` is a bit weird, at this point its a logging util
also the last setting of the interval will win, (also if we always use `defaultUpdateInterval` always then why pass it in 😅 )

Reply:
On that note, afaik, `upd` is a boolean. Assuming `defaultUpdateInterval` is refactored to a constant, it would seem prudent to combine the two remaining parameters into a struct, which may or may not also take `dkState`, and call Update with it or, if `dkState` is also in that struct, call it as a receiver function

Reply:
I agree that the interval parameter should be removed as it does not make sense. 

I think it's fine that it's kind of a logging util, as it adds the benefit that it sets the updated field for you on updates.

Reply:
For combining the 2 variables into a struct, what would be our advantage?

Reply:
> For combining the 2 variables into a struct, what would be our advantage?

increased read- and maintainability of the code base.
Also it's prudent to combine parameters into logical units, i.e., structs, and reduce parameter counts of functions.